ts/features/pn/api/backendPn.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  createFetchRequestForApi,
+  ApiHeaderJson,
+  composeHeaderProducers
+} from "italia-ts-commons/lib/requests";
+import * as r from "italia-ts-commons/lib/requests";
+import { SessionToken } from "../../../types/SessionToken";
+import { tokenHeaderProducer, withBearerToken } from "../../../utils/api";
+import { defaultRetryingFetch } from "../../../utils/fetch";
+import { ProblemJson } from "../../../../definitions/backend/ProblemJson";
+import { UpsertPNActivationT } from "../../../../definitions/pn/requestTypes";
+
+// TODO: remove this once codegen has been updated (https://pagopa.atlassian.net/browse/IA-914)
+//
+// The default decoder generated by codegen has a bug that doesn't handle
+// 204 with empty/undefined response (it always expects a JSON, so the decoding
+// fails). This bug has been fixed in a recend codegen release, but for now
+// we canno update it.
+export function upsertPNActivationDecoderCustom() {
+  return r.composeResponseDecoders(
+    r.composeResponseDecoders(
+      r.composeResponseDecoders(
+        r.constantResponseDecoder<undefined, 204>(204, undefined),
+        r.ioResponseDecoder<
+          400,
+          typeof ProblemJson["_A"],
+          typeof ProblemJson["_O"]
+        >(400, ProblemJson)
+      ),
+      r.constantResponseDecoder<undefined, 401>(401, undefined)
+    ),
+    r.ioResponseDecoder<
+      500,
+      typeof ProblemJson["_A"],
+      typeof ProblemJson["_O"]
+    >(500, ProblemJson)
+  );
+}
+
+const upsertActivation: UpsertPNActivationT = {
+  method: "post",
+  url: _ => `/api/v1/pn/activation`,
+  headers: composeHeaderProducers(tokenHeaderProducer, ApiHeaderJson),
+  query: query => ({ isTest: query.isTest }),
+  body: body => JSON.stringify(body.pNActivation),
+  response_decoder: upsertPNActivationDecoderCustom()
+};
+
+// client for handling PN endpoints
+export const BackendPnClient = (
```

Comment:
Why you are creating another client? You are still talking to the IO Backend.

Reply:
@francescopersico I preferred not to add feature-specific endpoints to the `BackendClient`. It's a pattern I found in-app (e.g. `BackendCGN` or `BackendCdcClient`).